### ISSUE TITLE
Соглашение о кодировании

### ISSUE THREAD

**Issue by dimio** [spacious-team/investbook#384]
Возможно, стоит закрепить соглашения на техническом уровне?
Могу предложить два варианта:
1. Подложить в проект xml-схему с описанием стиля и подключить [maven checkstyle plugin](https://maven.apache.org/plugins/maven-checkstyle-plugin/);
2. Ограничиться созданием файла настроек [.editorconfig](https://editorconfig.org/) (пример см. в прил.).
[editorconfig-example.zip](https://github.com/spacious-team/investbook/files/7683812/editorconfig-example.zip)

**Comment by vananiev** [comment]
Да, можно. У idea можно включить `.idea/codeStyles/codeStyleConfig.xml`. Только желательно все по дефолту настроить, чтобы у новичков с ненастроенной IDEA были минимальны расхождения.
1 и 2 - это альтернативные варианты?

**Comment by dimio** [comment]
Тогда мне стоит проверить, почему у меня idea не подхватила настройки форматирования оттуда, и пришлось править стиль.
Вар. 1 будет, скорее, дополнительным (и более трудозатратным) - настройки плагин, насколько я помню, может и идеевские использовать для проверки стиля.
Вар. 2 удобен тем, что этот конфиг - универсальный для различных редакторов/ide, если прописать в нём, по крайней мере, основное (отступы, кодировка, перевод строк и т.п.) - это будет работать во множестве поддерживающих конфиг редакторов (помимо idea).

**Comment by vananiev** [comment]
Давай попробуем подвести то что есть `.idea/codeStyles/`. Если  функционала не хватит (я сомневаюсь, там настроек много, например у команды Spring подход настройки проекта через этот файл), то используй из своих вариантов то, что тебе нравится. 

> idea не подхватила настройки форматирования
Может в идее должно быть настроено, чтобы использовал настройки проекта. а не глобальные.

**Comment by dimio** [comment]
> Давай попробуем подвести то что есть `.idea/codeStyles/`

Я полагал вопрос пока что закрытым на этом.
У себя включил использование схемы из проекта.

**Comment by vananiev** [comment]
ок
